sdk/storage/Azure.Storage.Blobs/tests/ContainerClientTests.cs

```diff
@@ -305,6 +310,44 @@ public async Task DeleteAsync_AccessConditions()
             }
         }
 
+        [Test]
+        public async Task DeleteAsync_AccessConditions_Conditions_IfMatch_Should_Throw()
+        {
+            var garbageLeaseId = GetGarbageLeaseId();
+            // Arrange
+            BlobServiceClient service = GetServiceClient_SharedKey();
+            BlobContainerClient container = InstrumentClient(service.GetBlobContainerClient(GetNewContainerName()));
+            await container.CreateAsync();
+            var conditions = new BlobRequestConditions()
+            {
+                IfMatch = new ETag("etag")
+            };
+
+            // Act
+            await TestHelper.AssertExpectedExceptionAsync<ArgumentOutOfRangeException>(
+                container.DeleteAsync(conditions: conditions),
+             e => { });
```

Comment:
Nit - indentation here and the test below.  You can also use TestHelper.CatchAsync if you don't want the lambda.